### ISSUE TITLE
Package request: MuJoCo Python bindings

### ISSUE THREAD

**Issue by tmplt** [NixOS/nixpkgs#262698]
**Project description**

Native Python bindings for the MuJoCo library.

**Metadata**

* homepage URL: https://mujoco.readthedocs.io/en/stable/python.html
* source URL: https://pypi.org/project/mujoco/
* license: apache 2
* platforms: linux

**Comment by tmplt** [comment]
I'm not used to packaging Python packages. A simple
```nix
buildPythonPackage rec {
    pname = "mujoco";
    version = "3.0.0";
    pyproject = false;

    src = pkgs.fetchPypi {
      inherit pname version;
      hash = "sha256-2ImKDfm9hCORaH7iou/Lp979VSCugCM/Gqnb/CkTeOw=";
    };
  }
```
fails with 
```
distutils.errors.DistutilsOptionError: No configuration found for dynamic 'entry-points'.
Some dynamic fields need to be specified via `tool.setuptools.dynamic`
others must be specified via the equivalent attribute in `setup.py`.
/nix/store/gv2cl6qvvslz5h15vqd89f1rpvrdg5yc-stdenv-linux/setup: line 1604: pop_var_context: head of shell_variables not a function context
```
during `setuptoolsBuildPhase`.

**Comment by tmplt** [comment]
@traversaro, I am working on an initial packaging with your system-wide dependency patches for MuJoCo. Thanks for those. I've reached as far as SdfLib currently and saw you did some CMake work there too. Is [your `conda` branch](https://github.com/traversaro/SdfLib/tree/conda) up-to-date, and are there any gotchas I should consider before looking into it?

**Comment by traversaro** [comment]
> @traversaro, I am working on an initial packaging with your system-wide dependency patches for MuJoCo. Thanks for those. I've reached as far as SdfLib currently and saw you did some CMake work there too. Is [your `conda` branch](https://github.com/traversaro/SdfLib/tree/conda) up-to-date, and are there any gotchas I should consider before looking into it?

Yes, it is kind of ready. That branch is meant to be used as done in https://github.com/traversaro/mujoco-conda-forge-ci/blob/usesystemdepsoptions2023sep/.github/workflows/ci.yml#L80, and with the further changes in mujoco in https://github.com/traversaro/mujoco/tree/usesystemdepsoptions2023sep .

**Comment by tmplt** [comment]
Good idea. I will do so as soon as I have a MWE and an overview of the whole dependency tree.

**Comment by tmplt** [comment]
Some packaging notes:
- mujoco's CMake scripts fetches deps because of pinned releases. Some downstream deps also do this. Some system-wide deps can be used, but not all (see above linked issue). Multiple APIs are used to check/fetch deps. Currently, these issues are patched in a fork.
-  Deps for Python bindings are all checked via the project's `findorfetch` script, which we can trivially patch. Ideally, the whole project should use the `findorfetch` API.
- PyPi tarball is impurely built/published and copies non-patched CMake files from project root, so these too must be patched. Solution: overwrite these files; merge Nix expressions in order to share patched project root.

**Comment by tmplt** [comment]
I have derivations that build at https://github.com/tmplt/nixpkgs/blob/3280b82643659959b74c8a51499bef28795b0a93/pkgs/applications/science/robotics/mujoco/default.nix. Before I PR I'm gonna see about pinning the last numerical dep (glm) and play around with the software a bit and hopefully stumble upon any eventual runtime issues.
